modules/samples/watertank/component/plugin/src/main/java/org/homeautomation/watertank/plugin/impl/DeviceTypeManagerService.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.homeautomation.watertank.plugin.impl;
+
+import org.homeautomation.watertank.plugin.constants.DeviceTypeConstants;
+import org.wso2.carbon.device.mgt.common.ProvisioningConfig;
+import org.wso2.carbon.device.mgt.common.DeviceManagementException;
+import org.wso2.carbon.device.mgt.common.DeviceManager;
+import org.wso2.carbon.device.mgt.common.app.mgt.ApplicationManager;
+import org.wso2.carbon.device.mgt.common.push.notification.PushNotificationConfig;
+import org.wso2.carbon.device.mgt.common.spi.DeviceManagementService;
+
+public class DeviceTypeManagerService implements DeviceManagementService {
+    private DeviceManager deviceManager;
+
+    @Override
+    public String getType() {
+        return DeviceTypeConstants.DEVICE_TYPE;
+    }
+
+    @Override
+    public void init() throws DeviceManagementException {
+        this.deviceManager = new DeviceTypeManager();
+    }
+
+    @Override
+    public DeviceManager getDeviceManager() {
+        return deviceManager;
+    }
+
+    @Override
+    public ApplicationManager getApplicationManager() {
+        return null;
```

Comment:
IoT device types do not have an application manager, so in here it is returning null.